src/MiniProfiler.AspNetCore/MiniProfilerExtensions.cs

```diff
@@ -15,6 +15,19 @@ public static class MiniProfilerExtensions
         // don't allocate this every time...
         private static readonly HtmlString includeNotFound = new HtmlString("<!-- Could not find 'include.partial.html' -->");
 
+        [ThreadStatic]
+        private static StringBuilder _stringBuilderCache;
+
+        private static StringBuilder GetStringBuilder() => _stringBuilderCache ?? CreateStringBuilder();
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static StringBuilder CreateStringBuilder()
+        {
+            var sb = new StringBuilder();
+            _stringBuilderCache = sb;
+            return sb;
+        }
```

Comment:
I agree with the approach (and happy to merge as-is). I'll likely share this elsewhere though, thoughts on me bringing https://github.com/opserver/Opserver/blob/overhaul/Opserver.Core/StringBuilderCache.cs into `MiniProfiler.Shared\Internal` after this is merged in?